### PR TITLE
Farshad#72 student notes

### PR DESCRIPTION
**Pull request by FBagdeli**

### CODE REVIEW

**Review by homonoviscoding — DISMISSED** [review]

**Review by homonoviscoding — DISMISSED** [review]

File: src/controllers/user.js [hunk 1]
```diff
@@ -32,15 +32,19 @@ export const create = async (req, res) => {
 
 export const getById = async (req, res) => {
   const id = parseInt(req.params.id)
+  const { role } = req.user
 
   try {
-    const foundUser = await User.findById(id)
-
+    let foundUser = {}
+    if (role === 'TEACHER') {
```
Comment (tzoltie) [inline src/controllers/user.js]: Create a `isTeacher = role === 'TEACHER'` variable that is checked in the if statement instead, reads better

File: src/controllers/user.js [hunk 2]
```diff
@@ -32,15 +32,19 @@ export const create = async (req, res) => {
...
+      foundUser = await User.findById(id, role)
+    }
     if (!foundUser) {
       return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
     }
-
-    return sendDataResponse(res, 200, foundUser)
+    return res.status(200).json({ status: 'success', data: foundUser })
   } catch (e) {
     return sendMessageResponse(res, 500, 'Unable to get user')
```
Comment (tzoltie) [inline src/controllers/user.js]: update the body of this error message so that it follows the same structure as the others, see line 45

**Review by homonoviscoding — DISMISSED** [review]

File: src/controllers/user.js [hunk 3]
```diff
@@ -32,24 +32,28 @@ export const create = async (req, res) => {
...
 export const getById = async (req, res) => {
   const id = parseInt(req.params.id)
+  const { role } = req.user
 
   try {
-    const foundUser = await User.findById(id)
-
+    let foundUser = {}
+    const isTeacher = 'TEACHER'
+    if (role === isTeacher) {
```
Comment (tzoltie) [inline src/controllers/user.js]: isTeacher should be `role === 'TEACHER'` then if the if statement will become `if (isTeacher)`

File: src/controllers/user.js [hunk 4]
```diff
@@ -32,24 +32,28 @@ export const create = async (req, res) => {
...
+      foundUser = await findByUniqueAsATeacher(id)
+    } else {
+      foundUser = await User.findById(id, role)
+    }
     if (!foundUser) {
       return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
     }
-
-    return sendDataResponse(res, 200, foundUser)
+    return res.status(200).json({ status: 'success', data: foundUser })
```
Comment (tzoltie) [inline src/controllers/user.js]: use the `sendDataResponse` function when returning the response body, see line 46 for what parameters are passed into the function

File: src/controllers/user.js [hunk 5]
```diff
@@ -37,15 +37,15 @@ export const getById = async (req, res) => {
   try {
     let foundUser = {}
     const isTeacher = 'TEACHER'
```
Comment (tzoltie) [inline src/controllers/user.js]: the role will need to be checked here, the if statement will always return false currently

File: src/controllers/user.js [hunk 6]
```diff
@@ -37,15 +37,15 @@ export const getById = async (req, res) => {
...
+    if (isTeacher) {
       foundUser = await findByUniqueAsATeacher(id)
     } else {
       foundUser = await User.findById(id, role)
     }
     if (!foundUser) {
       return sendDataResponse(res, 404, { error: ERR.USER_NOT_FOUND })
     }
-    return res.status(200).json({ status: 'success', data: foundUser })
+    return sendDataResponse(res, 201, foundUser)
```
Comment (tzoltie) [inline src/controllers/user.js]: `foundUser` should be stored in an object with a key of user

**Review by homonoviscoding — DISMISSED** [review]

**Review by MrStashy — DISMISSED** [review]

**Review by tzoltie — DISMISSED** [review]

**Review by homonoviscoding — DISMISSED** [review]

File: src/domain/user.js [hunk 7]
```diff
@@ -132,7 +138,11 @@ export default class User {
     return User._findByUnique('email', email)
   }
 
-  static async findById(id) {
+  static async findById(id, role) {
+    const isTeacher = role === 'TEACHER'
+    if (isTeacher) {
+      return _findByUniqueAsATeacher(id, role)
+    }
```
Comment (vherus) [inline src/domain/user.js:145]: I won't stop this from merging but a better solution here would be some optional parameters to include the relevant relations / fields. Something like `includeProfile = false, includeNotes = false`

**Review by vherus — DISMISSED** [review]

**Review by homonoviscoding — APPROVED** [review]

**Review by vherus — APPROVED** [review]

**Pull request merged**